firebase-vertexai/src/main/kotlin/com/google/firebase/vertexai/GenerativeModel.kt

```diff
@@ -79,7 +86,27 @@ internal constructor(
       apiKey,
       modelName,
       requestOptions.toInternal(),
-      "gl-kotlin/ fire/${BuildConfig.VERSION_NAME}"
+      "gl-kotlin/ fire/${BuildConfig.VERSION_NAME}",
+      object : HeaderProvider {
+        override val timeout: Duration
+          get() = 10.seconds
+
+        override suspend fun generateHeaders(): Map<String, String> {
+          if (appCheckTokenProvider == null) {
+            Log.w(TAG, "AppCheck not registered, skipping")
+            return emptyMap()
+          }
+          val mutableHeaderMap: MutableMap<String, String> = mutableMapOf()
+
+          val token = appCheckTokenProvider.getToken(false).await()
+          if (token.error != null) {
+            Log.w(TAG, "Error obtaining appcheck token", token.error)
+          }
+          mutableHeaderMap.put("X-Firebase-AppCheck", token.token)
+
+          return mutableHeaderMap
```

Comment:
There's no need for a mutable map here fwiw:

```suggestion
          val token = appCheckTokenProvider.getToken(false).await()
       
          if (token.error != null) {
            Log.w(TAG, "Error obtaining appcheck token", token.error)
            return emptyMap()
          }
          
          return mapOf("X-Firebase-AppCheck" to token.token)
```

Reply:
Thanks! that's a left over from previous testing. Fixed